### PR TITLE
Terminology change

### PR DESCRIPTION
**Pull request by itsdawei**
## TODO

- [x] Rename `n_solutions` with `batch_size` in Scheduler docstrings
- [x] Rename Optimizer to Scheduler
- [x] Rename `behavior (values)`, `bcs` to `measures`
- [x] Rename `objective_value` to `objective`
- [x] Fix visualize tests

## Status

- [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/icaros-usc/pyribs/blob/master/CONTRIBUTING.md)
- [x] I have formatted my code using `yapf`
- [x] I have tested my code by running `pytest`
- [x] I have linted my code with `pylint`
- [x] I have added a one-line description of my change to the changelog in `HISTORY.md`
- [x] This PR is ready to go

### CODE REVIEW

File: ribs/visualize.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def grid_archive_heatmap(archive,
         archive (GridArchive): A 2D :class:`~ribs.archives.GridArchive`.
         ax (matplotlib.axes.Axes): Axes on which to plot the heatmap.
             If ``None``, the current axis will be used.
-        transpose_bcs (bool): By default, the first BC in the archive will
+        transpose_measures (bool): By default, the first measure in the archive will
```
Comment (itsdawei) [inline ribs/visualize.py:140]: Not sure if this makes sense. Maybe "the first measure function in the archive"?

File: ribs/visualize.py [hunk 2]
```diff
@@ -319,7 +319,7 @@ def cvt_archive_heatmap(archive,
         plot_centroids (bool): Whether to plot the cluster centroids.
         plot_samples (bool): Whether to plot the samples used when generating
             the clusters.
-        transpose_bcs (bool): By default, the first BC in the archive will
+        transpose_measures (bool): By default, the first measure in the archive will
```
Comment (itsdawei) [inline ribs/visualize.py:322]: Same question as above

File: ribs/visualize.py [hunk 3]
```diff
@@ -490,7 +490,7 @@ def sliding_boundaries_archive_heatmap(archive,
             :class:`~ribs.archives.SlidingBoundariesArchive`.
         ax (matplotlib.axes.Axes): Axes on which to plot the heatmap.
             If ``None``, the current axis will be used.
-        transpose_bcs (bool): By default, the first BC in the archive will
+        transpose_measures (bool): By default, the first measure in the archive will
```
Comment (itsdawei) [inline ribs/visualize.py:493]: Same question as above

File: ribs/visualize.py [hunk 4]
```diff
@@ -631,14 +631,14 @@ def parallel_axes_plot(archive,
...
-            order specified does not need to have the same number of elements as
-            the number of behaviors in the archive, e.g. ``[1, 3]`` or
-            ``[1, 2, 3, 2]``.
+        measure_order (list of int or list of (int, str)): If this is a list
+            of ints, it specifies the axes order for measures (e.g. ``[2, 0,
+            1]``). If this is a list of tuples, each tuple takes the form
+            ``(int, str)`` where the int specifies the measure index and the str
+            specifies a name for the measure (e.g. ``[(1, "y-value"), (2,
+            "z-value"), (0, "x-value")]``). The order specified does not need
+            to have the same number of elements as the number of behaviors in
```
Comment (itsdawei) [inline ribs/visualize.py:640]: change behaviors to measure functions?

File: ribs/visualize.py [hunk 5]
```diff
@@ -663,8 +663,8 @@ def parallel_axes_plot(archive,
 
     Raises:
         ValueError: ``cbar_orientation`` has an invalid value.
-        ValueError: The bcs provided do not exist in the archive.
-        TypeError: ``bc_order`` is not a list of all ints or all tuples.
+        ValueError: The measures provided do not exist in the archive.
+        TypeError: ``measure_order`` is not a list of all ints or all tuples.
```
Comment (itsdawei) [inline ribs/visualize.py:667]: measure_order or measures_order

File: ribs/visualize.py [hunk 6]
```diff
@@ -675,32 +675,33 @@ def parallel_axes_plot(archive,
...
-            raise TypeError("bc_order must be a list of ints or a list of"
+            raise TypeError("measure_order must be a list of ints or a list of"
                             "tuples in the form (int, str)")
 
         if np.max(cols) >= archive.behavior_dim:
             raise ValueError(f"Invalid Behavior: requested behavior index "
                              f"{np.max(cols)}, but archive only has "
                              f"{archive.behavior_dim} behaviors.")
-        if any(bc < 0 for bc in cols):
+        if any(measure < 0 for measure in cols):
```
Comment (itsdawei) [inline ribs/visualize.py:704]: measures

File: tests/visualize_test.py [hunk 7]
```diff
@@ -118,7 +118,7 @@ def add_random_sphere(archive, x_range, y_range):
 #
 @pytest.fixture(scope="module")
 def grid_archive_1d():
-    """Deterministically created GridArchive with 1 BC."""
+    """Deterministically created GridArchive with 1 measure."""
```
Comment (itsdawei) [inline tests/visualize_test.py:121]: measures?

**Review by itsdawei — COMMENTED** [review]
Self-review; left some questions for @btjanaka to answer.

File: ribs/visualize.py (hunk 1 above)
Comment (btjanaka) [reply ribs/visualize.py:142]: I think this is fine as is.

File: ribs/visualize.py (hunk 5 above)
Comment (btjanaka) [reply ribs/visualize.py:669]: measure_order I think

File: ribs/visualize.py (hunk 1 above)
Comment (btjanaka) [reply ribs/visualize.py:142]: I think this is fine as is

File: tests/visualize_test.py (hunk 7 above)
Comment (btjanaka) [reply tests/visualize_test.py:121]: measure is fine

File: ribs/visualize.py (hunk 6 above)
Comment (btjanaka) [reply ribs/visualize.py:706]: measure. also, remove "Behavior" in the ValueError below this line.

File: ribs/visualize.py (hunk 4 above)
Comment (btjanaka) [reply ribs/visualize.py]: what you did is fine

File: ribs/visualize.py [hunk 8]
```diff
@@ -675,33 +677,34 @@ def parallel_axes_plot(archive,
                          f"but is '{cbar_orientation}'")
 
     # If there is no order specified, plot in increasing numerical order.
-    if bc_order is None:
+    if measure_order is None:
         cols = np.arange(archive.measure_dim)
-        axis_labels = [f"behavior_{i}" for i in range(archive.measure_dim)]
+        axis_labels = [f"measures_{i}" for i in range(archive.measure_dim)]
```
Comment (btjanaka) [inline ribs/visualize.py:682]: Please make the default be `measure_{i}`, then regenerate the plots.

File: ribs/visualize.py [hunk 9]
```diff
@@ -122,8 +124,8 @@ def grid_archive_heatmap(archive,
             >>> for x in np.linspace(-1, 1, 100):
             ...     for y in np.linspace(-1, 1, 100):
             ...         archive.add(solution=np.array([x,y]),
-            ...                     objective_value=-(x**2 + y**2),
-            ...                     behavior_values=np.array([x,y]))
+            ...                     objective=-(x**2 + y**2),
```
Comment (btjanaka) [inline ribs/visualize.py:127]: Most of these examples still do not work. Can you fix them to call `add` as a batch function in a followup PR?

**Review by btjanaka — CHANGES_REQUESTED** [review]

**Review by btjanaka — APPROVED** [review]

**Pull request merged**